includes/parserhooks/SetParserFunction.php

```diff
@@ -41,6 +46,15 @@ public function __construct( ParserData $parserData, MessageFormatter $messageFo
 	}
 
 	/**
+	 * @since  2.2
+	 *
+	 * @param Parser $parser
+	 */
+	public function setParser( $parser ) {
```

Comment:
Because I don't have to change the signature (means all test remain as is) and I only have to check for `$this->parser !== null` once.